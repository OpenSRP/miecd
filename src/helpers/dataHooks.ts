import {
    OPENSRP_SECURITY_AUTHENTICATE,
    LOCATION_FILTER_PARAM,
    RISK_CATEGORY_FILTER_PARAM,
    SMS_TYPE_FILTER_PARAM,
    SEARCH_FILTER_PARAM,
    PREGNANCY_DETECTION,
    UPDATE_PREGNANCY_DETECTION,
    NUTRITION_REGISTRATION,
    UPDATE_NUTRITION_REGISTRATION,
    DEATH_REPORT,
    UPDATE_DEATH_REPORT,
} from '../constants';
import { getQueryParams } from 'containers/LogFace/utils';
import { TFunction } from 'i18next';
import { useQuery } from 'react-query';
import { useSelector } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { OpenSRPService } from 'services/opensrp';
import { getSmsDataByFilters } from 'store/ducks/sms_events';
import { TreeNode } from './locationHierarchy/types';
import { generateJurisdictionTree } from './locationHierarchy/utils';
import { toastToError, getRiskCatFilter } from './utils';
import { LogFaceModules } from 'configs/settings';

const selectSmsData = getSmsDataByFilters();

/** hook to get this users assignment details, hierarchy, team assigned and location the team is assigned
 * @param t - the translator function
 */
export const useUserAssignment = (t: TFunction) => {
    return useQuery(
        'userAssignment',
        () => {
            const opensrpService = new OpenSRPService(OPENSRP_SECURITY_AUTHENTICATE);
            return opensrpService.read('');
        },
        {
            select: (response) => {
                return {
                    userHierarchy: generateJurisdictionTree(response.locations),
                    user: response.user,
                    team: response.team.team,
                    location: response.team.team.location,
                };
            },
            onError: (err: Error) => toastToError(err?.message ?? t('An error occurred')),
        },
    );
};

/** redux hok to select filtered sms data based on provided filters
 * @param location - route location details, to get search params
 * @param module - the module
 * @param userLocationTree - for filtering based on the user's assigned level
 */
export const useGetLogFaceFilteredSms = (
    location: RouteComponentProps['location'],
    module: LogFaceModules,
    userHierarchy?: TreeNode,
) => {
    return useSelector((state) => {
        const userLocationIdFilter = getQueryParams(location)[LOCATION_FILTER_PARAM] as string;
        const riskCategoryFilter = getQueryParams(location)[RISK_CATEGORY_FILTER_PARAM] as string;
        const smsTypeFilter = getQueryParams(location)[SMS_TYPE_FILTER_PARAM] as string;
        const searchFilter = getQueryParams(location)[SEARCH_FILTER_PARAM] as string;
        const matchesId = (node: TreeNode) => node.model.id === userLocationIdFilter;
        const userLocationNode = userHierarchy?.first(matchesId);
        let smsFilter = smsTypeFilter ? [smsTypeFilter] : undefined;
        switch (smsTypeFilter) {
            case PREGNANCY_DETECTION:
                smsFilter = [PREGNANCY_DETECTION, UPDATE_PREGNANCY_DETECTION];
                break;
            case NUTRITION_REGISTRATION:
                smsFilter = [NUTRITION_REGISTRATION, UPDATE_NUTRITION_REGISTRATION];
                break;
            case DEATH_REPORT:
                smsFilter = [DEATH_REPORT, UPDATE_DEATH_REPORT];
                break;
            default:
                break;
        }
        return selectSmsData(state, {
            locationNode: userLocationNode,
            riskCategory: getRiskCatFilter(module, riskCategoryFilter),
            smsTypes: smsFilter,
            searchFilter,
            module,
        });
    });
};
