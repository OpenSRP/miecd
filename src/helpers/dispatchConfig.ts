import { ConfigState, setAllConfigs, getAllConfigs } from '@opensrp/pkg-config';
import { OPENSRP_REST_API_BASE_URL } from 'configs/settings';
import { BACKEND_ACTIVE } from '../configs/env';
import { BACKEND_LOGIN_URL, REACT_LOGIN_URL } from '../constants';
import i18n from '../mls';

export const APP_LOGIN_URL = BACKEND_ACTIVE ? BACKEND_LOGIN_URL : REACT_LOGIN_URL;

const defaultvalues = getAllConfigs();

const configObject: ConfigState = {
    ...defaultvalues,
    appLoginURL: APP_LOGIN_URL,
    opensrpBaseURL: OPENSRP_REST_API_BASE_URL,
    i18n,
};

setAllConfigs(configObject);
