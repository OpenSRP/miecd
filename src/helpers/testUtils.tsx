import { mount, MountRendererProps } from 'enzyme';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import i18n from '../mls';
import { act } from 'react-dom/test-utils';

export const mountWithTranslations: typeof mount = (component: React.ReactNode, options?: MountRendererProps) => {
    return mount(<I18nextProvider i18n={i18n}>{component}</I18nextProvider>, options);
};

/**
 * Utility to await promises to resolve in tests
 * @param ms time in ms
 */
export async function waitForPromises() {
    await act(async () => {
        await new Promise((res) => setImmediate(res));
    });
}
