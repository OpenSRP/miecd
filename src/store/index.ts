import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import { Dictionary } from '@onaio/utils';

/** Initial reducers in the reducer registry */
const defaultReducers: Dictionary = {
    router: connectReducer /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
    store.replaceReducer(combine(reducers));
});

export default store;
