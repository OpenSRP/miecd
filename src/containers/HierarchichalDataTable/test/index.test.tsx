/* eslint-disable @typescript-eslint/no-explicit-any */
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Provider } from 'react-redux';
import { Route, MemoryRouter } from 'react-router';
import ConnectedHierarchicalDataTable from '..';
import { mountWithTranslations } from '../../../helpers/testUtils';
import store from '../../../store/index';
import {
    villages,
    communes,
    districts,
    provinces,
    pregnancySmsData,
    nutritionSmsData,
    nbcPncSmsData,
} from '../../HierarchichalDataTable/test/fixtures';
import { authenticateUser } from '@onaio/session-reducer';
import { getOpenSRPUserInfo } from '@onaio/gatekeeper';
import { QueryClient, QueryClientProvider } from 'react-query';
import fetchMock from 'fetch-mock';
import flushPromises from 'flush-promises';
import { act } from 'react-dom/test-utils';

jest.mock('@fortawesome/react-fontawesome');

interface RenderWithProviders {
    store: typeof store;
    queryClient: QueryClient;
    initialEntry: string;
}
// Render with Providers Helper function
function renderWithProviders(UI: React.FC, { store, queryClient, initialEntry }: RenderWithProviders) {
    return mountWithTranslations(
        <Provider store={store}>
            <QueryClientProvider client={queryClient}>
                <MemoryRouter initialEntries={[initialEntry]}>
                    <Route
                        exact
                        path="/:module?/:risk_highlighter?/:title?/:current_level?/:direction?/:node_id?/:permission_level?/:from_level?"
                        component={UI}
                    />
                </MemoryRouter>
            </QueryClientProvider>
        </Provider>,
    );
}

describe('HierarchichalDataTable', () => {
    beforeAll(() => {
        const OpenSRPAPIResponse = {
            oAuth2Data: {
                access_token: 'hunter2',
                expires_in: '3599',
                state: 'opensrp',
                token_type: 'bearer',
            },
            preferredName: 'Superset User',
            roles: ['Provider'],
            username: 'admin',
        };
        const { authenticated, user, extraData } = getOpenSRPUserInfo(OpenSRPAPIResponse);
        store.dispatch(authenticateUser(authenticated, user, extraData));
    });

    beforeEach(() => {
        fetchMock
            .get(`https://discover.miecd-stage.smartregister.org/oauth-authorized/opensrp`, {})
            .get(`https://discover.miecd-stage.smartregister.org/superset/slice_json/12`, villages)
            .get(`https://discover.miecd-stage.smartregister.org/superset/slice_json/13`, communes)
            .get(`https://discover.miecd-stage.smartregister.org/superset/slice_json/10`, districts)
            .get(`https://discover.miecd-stage.smartregister.org/superset/slice_json/11`, provinces);
    });

    afterEach(() => {
        jest.resetAllMocks();
        jest.clearAllMocks();
        fetchMock.restore();
    });

    it('renders without crashing', () => {
        const queryClient = new QueryClient();

        const wrapper = shallow(
            <Provider store={store}>
                <QueryClientProvider client={queryClient}>
                    <ConnectedHierarchicalDataTable />
                </QueryClientProvider>
            </Provider>,
        );

        // show ripple loader
        expect(wrapper.find('Ripple')).toBeTruthy();

        wrapper.unmount();
    });

    it('renders correctly for Pregnancy module', async () => {
        const queryClient = new QueryClient();

        // fetch pregnancy sms slice
        fetchMock.get(`https://discover.miecd-stage.smartregister.org/superset/slice_json/8`, pregnancySmsData);

        // mount for pregnancy module with 'all' risk type
        let wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry: '/Pregnancy/all/11 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // show ripple loader
        expect(wrapper.find('Ripple')).toBeTruthy();

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // expect no loader
        expect(wrapper.find('Ripple').exists()).toBeFalsy();

        // show's hierarchy table
        expect(toJson(wrapper.find('Table'))).toMatchSnapshot('hierarchicalDataTable');

        // mount for pregnancy module with 'red' risk type
        wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry: '/Pregnancy/red/11 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // show's hierarchy table with red alert highlighted

        // table rows with 'red-alert' class
        const redAlertTr = wrapper.find('Table tr').find('.red-alert');
        // expect four rows (3 provinces and totals)
        expect(redAlertTr).toHaveLength(4);

        // mount for pregnancy module with 'risk' risk type
        wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry: '/Pregnancy/risk/11 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // show's hierarchy table with red alert highlighted

        // table rows with 'risk' class
        const riskTd = wrapper.find('Table tr').find('.risk');
        // expect four rows (3 provinces and totals)
        expect(riskTd).toHaveLength(4);

        // mount for pregnancy module with 'no' risk type
        wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry:
                '/Pregnancy/no/0 Total Pregnancies due in 2 weeks/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // show's hierarchy table with no_risk highlighted

        // table rows with 'risk' class
        const noRiskTd = wrapper.find('Table tr').find('.no');
        // expect four rows (3 provinces and totals)
        expect(noRiskTd).toHaveLength(4);

        wrapper.unmount();
    });

    it('renders correctly for NBC_AND_PNC module', async () => {
        const queryClient = new QueryClient();

        // fetch nbcPnc sms slice
        fetchMock.get(`https://discover.miecd-stage.smartregister.org/superset/slice_json/5`, nbcPncSmsData);

        // mount for nbcPnc module with 'all' risk type
        let wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry: '/NBC & PNC_CHILD/all/4 Total Newborns/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // show ripple loader
        expect(wrapper.find('Ripple')).toBeTruthy();

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // expect no loader
        expect(wrapper.find('Ripple').exists()).toBeFalsy();

        // show's hierarchy table
        expect(toJson(wrapper.find('Table'))).toMatchSnapshot('hierarchicalDataTable');

        // mount for nbcPnc module with 'red' risk type
        wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry: '/NBC & PNC_WOMAN/red/3 Total Mothers in PNC/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // show's hierarchy table with red alert highlighted

        // table rows with 'red-alert' class
        const redAlertTr = wrapper.find('Table tr').find('.red-alert');
        // expect four rows (3 provinces and totals)
        expect(redAlertTr).toHaveLength(4);

        // mount for nbcPnc module with 'risk' risk type
        wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry: '/NBC & PNC_CHILD/risk/4 Total Newborns/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // show's hierarchy table with red alert highlighted

        // table rows with 'risk' class
        const riskTd = wrapper.find('Table tr').find('.risk');
        // expect four rows (3 provinces and totals)
        expect(riskTd).toHaveLength(4);

        // mount for nbcPnc module with 'no' risk type
        wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry: '/NBC & PNC_CHILD/no/4 Total Newborns/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // show's hierarchy table with no_risk highlighted

        // table rows with 'no' class
        const noRiskTd = wrapper.find('Table tr').find('.no');
        // expect four rows (3 provinces and totals)
        expect(noRiskTd).toHaveLength(4);

        wrapper.unmount();
    });

    it('renders correctly for Nutrition module', async () => {
        const queryClient = new QueryClient();

        // fetch Nutrition sms slice
        fetchMock.get(`https://discover.miecd-stage.smartregister.org/superset/slice_json/6`, nutritionSmsData);

        // mount for pregnancy module with 'all' risk type
        let wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry:
                '/Nutrition/all/11%20Total%20Children%20Under%205/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // show ripple loader
        expect(wrapper.find('Ripple')).toBeTruthy();

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // expect no loader
        expect(wrapper.find('Ripple').exists()).toBeFalsy();

        // show's hierarchy table
        expect(toJson(wrapper.find('Table'))).toMatchSnapshot('hierarchicalDataTable');

        // mount for Nutrition module with 'stunted' risk type
        wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry: '/Nutrition/stunted/11 Total Children Under 5/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // show's hierarchy table with stunted highlighted

        // table rows with 'stunted' class
        const stuntedTr = wrapper.find('Table tr').find('.stunted');
        // expect four rows (3 provinces and totals)
        expect(stuntedTr).toHaveLength(4);

        // mount for Nutrition module with 'severe wasting' risk type
        wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry:
                '/Nutrition/severe wasting/9 Total Children Under 2/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // show's hierarchy table with severe wasting highlighted

        // table rows with 'severe-wasting' class
        const severeWastingTd = wrapper.find('Table tr').find('.severe-wasting');
        // expect four rows (3 provinces and totals)
        expect(severeWastingTd).toHaveLength(4);

        // mount for Nutrition module with 'overweight' risk type
        wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry:
                '/Nutrition/overweight/9 Total Children Under 2/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // show's hierarchy table with overweight highlighted

        // table rows with 'overweight' class
        const overweight = wrapper.find('Table tr').find('.overweight');
        // expect four rows (3 provinces and totals)
        expect(overweight).toHaveLength(4);

        // mount for Nutrition module with 'inappropriately fed' risk type
        wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry:
                '/Nutrition/inappropriately fed/11 Total Children Under 5/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // show's hierarchy table with inappropriately fed highlighted

        // table rows with 'inappropriately-fed' class
        const inappropriatelyFed = wrapper.find('Table tr').find('.inappropriately-fed');
        // expect four rows (3 provinces and totals)
        expect(inappropriatelyFed).toHaveLength(4);

        // mount for Nutrition module with 'normal' risk type
        wrapper = renderWithProviders(ConnectedHierarchicalDataTable, {
            store,
            queryClient,
            initialEntry: '/Nutrition/normal/11 Total Children Under 5/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0',
        });

        // wait for fetches to complete
        await act(async () => {
            await flushPromises();
            wrapper.update();
        });

        // show's hierarchy table with normal highlighted

        // table rows with 'inappropriately-fed' class
        const normal = wrapper.find('Table tr').find('.normal');
        // expect four rows (3 provinces and totals)
        expect(normal).toHaveLength(4);

        wrapper.unmount();
    });
});
