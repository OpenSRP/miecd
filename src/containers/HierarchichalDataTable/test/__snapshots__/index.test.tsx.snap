// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HierarchichalDataTable must show the no record message when there's no data to show for a commune: norecord 1`] = `null`;

exports[`HierarchichalDataTable renders correctly: hierarchichalDataTable 1`] = `
<Table
  borderless={true}
  responsiveTag="div"
  striped={true}
  tag="table"
>
  <table
    className="table table-borderless table-striped"
  >
    <thead
      id="header"
    >
      <tr>
        <th
          className="default-width"
        />
        <th
          className="default-width"
        >
          Red Alert
        </th>
        <th
          className="default-width"
        >
          Risk
        </th>
        <th
          className="default-width"
        >
          No Risk
        </th>
        <th
          className="default-width totals"
        >
          Total
        </th>
      </tr>
    </thead>
    <tbody
      id="body"
    >
      <tr
        className="cursor-pointer color-blue"
        key="d79168ed-ae95-43d1-8e45-9078212815f6"
        onClick={[Function]}
      >
        <td
          className="default-width"
        >
          Lang Son Province
        </td>
        <td
          className="default-width "
        >
          0
        </td>
        <td
          className="default-width "
        >
          0
        </td>
        <td
          className="default-width "
        >
          0
        </td>
        <td
          className="default-width totals"
        >
          0
        </td>
      </tr>
      <tr
        className="cursor-pointer color-blue"
        key="c13b5c95-55c3-4848-9509-b82596274d96"
        onClick={[Function]}
      >
        <td
          className="default-width"
        >
          Dien Bien
        </td>
        <td
          className="default-width "
        >
          0
        </td>
        <td
          className="default-width "
        >
          0
        </td>
        <td
          className="default-width "
        >
          0
        </td>
        <td
          className="default-width totals"
        >
          0
        </td>
      </tr>
      <tr
        className="cursor-pointer color-blue"
        key="78a12165-3c12-471f-8755-c96bac123292"
        onClick={[Function]}
      >
        <td
          className="default-width"
        >
          Hanoi
        </td>
        <td
          className="default-width "
        >
          10
        </td>
        <td
          className="default-width "
        >
          9
        </td>
        <td
          className="default-width "
        >
          6
        </td>
        <td
          className="default-width totals"
        >
          25
        </td>
      </tr>
      <tr
        className="cursor-pointer color-blue"
        key="f7d1d9bb-95dc-43bc-8434-0dddd1f3d374"
        onClick={[Function]}
      >
        <td
          className="default-width"
        >
          Loc Binh
        </td>
        <td
          className="default-width "
        >
          0
        </td>
        <td
          className="default-width "
        >
          0
        </td>
        <td
          className="default-width "
        >
          0
        </td>
        <td
          className="default-width totals"
        >
          0
        </td>
      </tr>
      <tr
        className="totals-row"
        key="total"
      >
        <td
          className="default-width"
          id="total"
        >
          Total (Province)
        </td>
        <td
          className="default-width "
        >
          10
        </td>
        <td
          className="default-width "
        >
          9
        </td>
        <td
          className="default-width "
        >
          6
        </td>
        <td
          className="default-width"
        >
          25
        </td>
      </tr>
    </tbody>
  </table>
</Table>
`;

exports[`HierarchichalDataTable renders correctly: hierarchichalDataTable 2`] = `
<tbody
  id="body"
>
  <tr
    id="no-rows"
  >
    <td>
      -(
    </td>
  </tr>
  <tr
    className="totals-row"
    key="total"
  >
    <td
      className="default-width"
      id="total"
    >
      Total (District)
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width"
    >
      0
    </td>
  </tr>
</tbody>
`;

exports[`HierarchichalDataTable renders correctly: hierarchichalDataTable 3`] = `
<tbody
  id="body"
>
  <tr
    id="no-rows"
  >
    <td>
      -(
    </td>
  </tr>
  <tr
    className="totals-row"
    key="total"
  >
    <td
      className="default-width"
      id="total"
    >
      Total (Commune)
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width"
    >
      0
    </td>
  </tr>
</tbody>
`;

exports[`HierarchichalDataTable renders correctly: hierarchichalDataTable 4`] = `
<tbody
  id="body"
>
  <tr
    id="no-rows"
  >
    <td>
      -(
    </td>
  </tr>
  <tr
    className="totals-row"
    key="total"
  >
    <td
      className="default-width"
      id="total"
    >
      Total (Village)
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width"
    >
      0
    </td>
  </tr>
</tbody>
`;

exports[`HierarchichalDataTable renders correctly: hierarchichalDataTable 5`] = `
<tbody
  id="body"
>
  <tr
    id="no-rows"
  >
    <td>
      -(
    </td>
  </tr>
  <tr
    className="totals-row"
    key="total"
  >
    <td
      className="default-width"
      id="total"
    >
      Total (Commune)
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width"
    >
      0
    </td>
  </tr>
</tbody>
`;

exports[`HierarchichalDataTable renders correctly: hierarchichalDataTable 6`] = `
<tbody
  id="body"
>
  <tr
    id="no-rows"
  >
    <td>
      -(
    </td>
  </tr>
  <tr
    className="totals-row"
    key="total"
  >
    <td
      className="default-width"
      id="total"
    >
      Total (District)
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width"
    >
      0
    </td>
  </tr>
</tbody>
`;

exports[`HierarchichalDataTable renders correctly: hierarchichalDataTable 7`] = `
<tbody
  id="body"
>
  <tr
    className="cursor-pointer color-blue"
    key="d79168ed-ae95-43d1-8e45-9078212815f6"
    onClick={[Function]}
  >
    <td
      className="default-width"
    >
      Lang Son Province
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width totals"
    >
      0
    </td>
  </tr>
  <tr
    className="cursor-pointer color-blue"
    key="c13b5c95-55c3-4848-9509-b82596274d96"
    onClick={[Function]}
  >
    <td
      className="default-width"
    >
      Dien Bien
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width totals"
    >
      0
    </td>
  </tr>
  <tr
    className="cursor-pointer color-blue"
    key="78a12165-3c12-471f-8755-c96bac123292"
    onClick={[Function]}
  >
    <td
      className="default-width"
    >
      Hanoi
    </td>
    <td
      className="default-width "
    >
      10
    </td>
    <td
      className="default-width "
    >
      9
    </td>
    <td
      className="default-width "
    >
      6
    </td>
    <td
      className="default-width totals"
    >
      25
    </td>
  </tr>
  <tr
    className="cursor-pointer color-blue"
    key="f7d1d9bb-95dc-43bc-8434-0dddd1f3d374"
    onClick={[Function]}
  >
    <td
      className="default-width"
    >
      Loc Binh
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width totals"
    >
      0
    </td>
  </tr>
  <tr
    className="totals-row"
    key="total"
  >
    <td
      className="default-width"
      id="total"
    >
      Total (Province)
    </td>
    <td
      className="default-width "
    >
      10
    </td>
    <td
      className="default-width "
    >
      9
    </td>
    <td
      className="default-width "
    >
      6
    </td>
    <td
      className="default-width"
    >
      25
    </td>
  </tr>
</tbody>
`;

exports[`HierarchichalDataTable renders correctly: hierarchichalDataTable 8`] = `
<tbody
  id="body"
>
  <tr
    id="no-rows"
  >
    <td>
      -(
    </td>
  </tr>
  <tr
    className="totals-row"
    key="total"
  >
    <td
      className="default-width"
      id="total"
    >
      Total (District)
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width"
    >
      0
    </td>
  </tr>
</tbody>
`;

exports[`HierarchichalDataTable renders correctly: hierarchichalDataTable 9`] = `
<tbody
  id="body"
>
  <tr
    className="cursor-pointer color-blue"
    key="d79168ed-ae95-43d1-8e45-9078212815f6"
    onClick={[Function]}
  >
    <td
      className="default-width"
    >
      Lang Son Province
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width totals"
    >
      0
    </td>
  </tr>
  <tr
    className="cursor-pointer color-blue"
    key="c13b5c95-55c3-4848-9509-b82596274d96"
    onClick={[Function]}
  >
    <td
      className="default-width"
    >
      Dien Bien
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width totals"
    >
      0
    </td>
  </tr>
  <tr
    className="cursor-pointer color-blue"
    key="78a12165-3c12-471f-8755-c96bac123292"
    onClick={[Function]}
  >
    <td
      className="default-width"
    >
      Hanoi
    </td>
    <td
      className="default-width "
    >
      10
    </td>
    <td
      className="default-width "
    >
      9
    </td>
    <td
      className="default-width "
    >
      6
    </td>
    <td
      className="default-width totals"
    >
      25
    </td>
  </tr>
  <tr
    className="cursor-pointer color-blue"
    key="f7d1d9bb-95dc-43bc-8434-0dddd1f3d374"
    onClick={[Function]}
  >
    <td
      className="default-width"
    >
      Loc Binh
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width "
    >
      0
    </td>
    <td
      className="default-width totals"
    >
      0
    </td>
  </tr>
  <tr
    className="totals-row"
    key="total"
  >
    <td
      className="default-width"
      id="total"
    >
      Total (Province)
    </td>
    <td
      className="default-width "
    >
      10
    </td>
    <td
      className="default-width "
    >
      9
    </td>
    <td
      className="default-width "
    >
      6
    </td>
    <td
      className="default-width"
    >
      25
    </td>
  </tr>
</tbody>
`;
